.gitignore

```diff
@@ -2,4 +2,5 @@
 !.gitignore
 !.htaccess
 !.gitkeep
-vendor/*
\ No newline at end of file
+vendor/*
+composer.lock
```

Comment:
I know it's recommended to keep composer.lock versioned in the project. What is the reason to remove it?

Reply:
I'm also torn about whether plugins should have composer.lock or not. The fact that no other plugins do not have it is not that relevant as it was useless until the Mautic Marketplace enabled to install plugins with Composer dependencies. I'm hoping for pros and cons discussion so I can get my opinion straight :)

The con would be that composer.lock is too strict so it may get hard to install it together with other plugins. 

The pro would be that if the CI tests will pass then the plugin dependencies are locked and won't surprise anyone that it cause troubles with other versions of the dependencies.

Reply:
> Inside composer documentation : https://getcomposer.org/doc/02-libraries.md#lock-file
> For your library you may commit the composer.lock file if you want to. This can help your team to always test against the same dependency versions. However, this lock file will not have any effect on other projects that depend on it. It only has an effect on the main project.